### PR TITLE
CI: add `go mod tidy` workflow

### PR DESCRIPTION
**Pull request by DentonGentry**
Fixes https://github.com/tailscale/tailscale/issues/4567

Signed-off-by: Denton Gentry <dgentry@tailscale.com>

### CODE REVIEW

File: .github/workflows/go_mod_tidy.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: go generate
```
Comment (bradfitz) [inline .github/workflows/go_mod_tidy.yml:1]: Name

File: .github/workflows/go_mod_tidy.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  cancel-in-progress: true
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
```
Comment (bradfitz) [inline .github/workflows/go_mod_tidy.yml:22]: Use version-file or whatever like corp?

Comment (DentonGentry) [reply .github/workflows/go_mod_tidy.yml]: Done

File: .github/workflows/go_mod_tidy.yml (hunk 1 above)
Comment (DentonGentry) [reply .github/workflows/go_mod_tidy.yml]: Updated

**Review by bradfitz — APPROVED** [review]

**Pull request merged**